iXBRLViewerPlugin/iXBRLViewer.py

```diff
@@ -351,17 +352,19 @@ def createViewer(self, scriptUrl="js/dist/ixbrlviewer.js", showValidations = Tru
 
         if dts.modelDocument.type == Type.INLINEXBRLDOCUMENTSET:
             # Sort by object index to preserve order in which files were specified.
-            docSet = sorted(dts.modelDocument.referencesDocument.keys(), key=lambda x: x.objectIndex)
-            docSetFiles = list(map(lambda x: os.path.basename(x.filepath), docSet))
-            self.taxonomyData["docSetFiles"] = docSetFiles
+            xmlDocsByFilename = {
+                os.path.basename(doc.filepath): deepcopy(doc.xmlDocument)
+                for doc in sorted(dts.modelDocument.referencesDocument.keys(), key=lambda x: x.objectIndex)
+            }
+            self.taxonomyData["docSetFiles"] = list(xmlDocsByFilename.values())
 
-            for n in range(0, len(docSet)):
-                iv.addFile(iXBRLViewerFile(docSetFiles[n], docSet[n].xmlDocument))
+            for filename, xmlDocument in xmlDocsByFilename.items():
+                iv.addFile(iXBRLViewerFile(filename, xmlDocument))
 
-            xmlDocument = docSet[0].xmlDocument 
+            xmlDocument = next(iter(xmlDocsByFilename.values()))
 
         else:
-            xmlDocument = dts.modelDocument.xmlDocument
+            xmlDocument = deepcopy(dts.modelDocument.xmlDocument)
```

Comment:
What are these changes for?

Reply:
Hopefully the comment opens since its resolved https://github.com/Workiva/ixbrl-viewer/pull/371#discussion_r1095187534 but since the plugin opens on startup it modifies `dts.modelDocument.xmlDocument` to have the ixbrlviewer json and javascript added. Then when you go to save there is a check to make sure you aren't modifying a document with an ixbrl-viewer (which you always are) and thus prevents you from saving. The deepcopy lets you create an ixbrl-viewer as many times as you want from the same document without ever modifying the original source document which is important since the save functionality allows you to supply a URL

Reply:
Ah yes.  That makes sense, thank you.